src/core/database.py

```diff
@@ -235,6 +239,15 @@ def _createFieldsForFollowerTable(self) -> QgsFields:
 
         return fields
 
+    def _createFieldsForGpsTable(self) -> QgsFields:
+        fields = QgsFields()
+        fields.append(QgsField("Latitude", QVariant.Double))
+        fields.append(QgsField("Longitude", QVariant.Double))
```

Comment:
latitude and longitude don't have be stored as fields. Instead, we need a point geometry type on the layer.